tests/vip-integrations/test-vip-integrations.php

```diff
@@ -3,52 +3,85 @@
 namespace Automattic\VIP\Integrations;
 
 use WP_UnitTestCase;
+use InvalidArgumentException;
 
 require_once __DIR__ . '/fake-integration.php';
 
-/**
- * @runTestsInSeparateProcesses
- * @preserveGlobalState disabled
- */
 class VIP_Integrations_Test extends WP_UnitTestCase {
 	public function setUp(): void {
-		// Remove existing default integrations to avoid conflicts when re-registering
-		Integrations::instance()->remove_registered();
+		parent::setUp();
+
+		// Reset static is_loaded status on integration so it can be reloaded
+		FakeIntegration::$is_loaded = false;
 	}
 
-	public function test_register_integration_as_class_name() {
-		add_action( 'register_vip_integrations', function() {
-			Integrations::instance()->register( 'fake-integration', FakeIntegration::class );
-		});
+	// Registration
+
+	public function test_register_integration_as_class_name_loads() {
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration-class', FakeIntegration::class );
 
-		activate( 'fake-integration' );
+		$integrations->activate( 'fake-integration-class' );
 
-		$this->run_integration_actions();
-		$this->assertTrue( FakeIntegration::$is_integrated );
+		$integrations->load_active();
+		$this->assertTrue( FakeIntegration::$is_loaded );
 	}
 
-	public function test_register_integration_as_instantiated_class() {
+	public function test_register_integration_as_instantiated_class_loads() {
 		$fake_integration = new class() extends Integration {
-			public $is_integrated = false;
+			public $is_loaded = false;
 
-			public function integrate( array $config ): void {
-				$this->is_integrated = true;
+			public function load( array $config ): void {
+				$this->is_loaded = true;
 			}
 		};
 
-		add_action( 'register_vip_integrations', function() use ( $fake_integration ) {
-			Integrations::instance()->register( 'fake-integration', $fake_integration );
-		});
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration-instance', $fake_integration );
 
-		activate( 'fake-integration' );
+		$integrations->activate( 'fake-integration-instance' );
 
-		$this->run_integration_actions();
-		$this->assertTrue( $fake_integration->is_integrated );
+		$integrations->load_active();
+		$this->assertTrue( $fake_integration->is_loaded );
 	}
 
-	private function run_integration_actions() {
-		do_action( 'register_vip_integrations' );
-		do_action( 'activate_vip_integrations' );
-		do_action( 'integrate_vip_integrations' );
+	// Non-activation
+
+	public function test_register_integration_without_activation_does_not_load() {
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration', FakeIntegration::class );
+
+		$integrations->load_active();
+		$this->assertFalse( FakeIntegration::$is_loaded );
+	}
+
+	// Errors
+
+	public function test_double_slug_registration_throws() {
```

Comment:
Consider adding the exception name to the test name:

```suggestion
	public function test_double_slug_registration_throws_invalidArgumentException() {
```

Tip: Run tests with `--testdox` to see how nicer test method names can help make them more descriptive.

Reply:
Good tip! I haven't used that before. I'll use a more verbose `test__<action>__<expected-outcome>` format for these.